diff --git a/react-native.d.ts b/react-native.d.ts
index cce0d42..741d245 100644
--- a/react-native.d.ts
+++ b/react-native.d.ts
@@ -516,6 +516,10 @@ type LegendListRef$1 = {
      * Returns the internal state of the scroll virtualization.
      */
     getState(): LegendListState$1;
+    /**
+     * Updates a measured item size and recalculates list positions as needed.
+     */
+    updateItemSize(itemKey: string, size: Pick<LayoutRectangle, "height" | "width">): void;
     /**
      * Scrolls a specific index into view.
      * @param params - Parameters for scrolling.
diff --git a/react-native.js b/react-native.js
index 8e6c503..2397600 100644
--- a/react-native.js
+++ b/react-native.js
@@ -5882,6 +5882,7 @@ function createImperativeHandle(ctx, scheduleImperativeScrollCommit) {
     getNativeScrollRef: () => refScroller.current,
     getScrollableNode: () => refScroller.current.getScrollableNode(),
     getScrollResponder: () => refScroller.current.getScrollResponder(),
+    updateItemSize: (itemKey, sizeObj) => updateItemSize(ctx, itemKey, sizeObj),
     getState: () => ({
       activeStickyIndex: peek$(ctx, "activeStickyIndex"),
       contentLength: getContentSize(ctx),
diff --git a/react-native.mjs b/react-native.mjs
index 6e57ebb..6c12e36 100644
--- a/react-native.mjs
+++ b/react-native.mjs
@@ -5861,6 +5861,7 @@ function createImperativeHandle(ctx, scheduleImperativeScrollCommit) {
     getNativeScrollRef: () => refScroller.current,
     getScrollableNode: () => refScroller.current.getScrollableNode(),
     getScrollResponder: () => refScroller.current.getScrollResponder(),
+    updateItemSize: (itemKey, sizeObj) => updateItemSize(ctx, itemKey, sizeObj),
     getState: () => ({
       activeStickyIndex: peek$(ctx, "activeStickyIndex"),
       contentLength: getContentSize(ctx),
